app/models/reports/performance_score.rb

```diff
@@ -0,0 +1,49 @@
+module Reports
+  class PerformanceScore
+    CONTROL_SCORE_WEIGHT = 0.5
+    VISITS_SCORE_WEIGHT = 0.3
+    REGISTRATIONS_SCORE_WEIGHT = 0.2
+    TARGET_REGISTRATION_RATE = 0.1
+
+    def initialize(region:, result:)
+      @region = region
+      @result = result
+    end
+
+    def overall_score
+      # max score: 100
+      (control_score + visits_score + registrations_score)
+    end
+
+    def control_score
+      CONTROL_SCORE_WEIGHT * @result.controlled_patients_rate.values.last
```

Comment:
Could you explain the significance of `controlled_patients_rate.values.last`? What does it mean to pull the last `controlled_patients_rate` value out?

Also, it seems like the `Result` object has a method called [last_value](https://github.com/simpledotorg/simple-server/blob/master/app/models/reports/result.rb#L51) which seems like it does the same thing  that you could use (although I don't understand the meaning behind that either).

Reply:
the `Result` object contains a hash of values keyed by dates, so you can get the control rate for May 2019, for example. `values.last` returns the most recent value. I suppose switching to a more consistent method may make sense for this (to cover for weirdness in hash sorting, for example). What say you, @rsanheim? Regardless, that's a separate PR.